internal/user/user.go

```diff
@@ -0,0 +1,58 @@
+package user
+
+import (
+	"context"
+	"errors"
+	"time"
+)
+
+type User struct {
```

Comment:
We are moving these common structs to `model` pkg. 
cc: @AkarshSatija @ravisuhag